Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift

```diff
@@ -265,14 +242,10 @@ AppDelegateViewModelOutputs {
       .filter { AppEnvironment.current.currentUser != nil }
 
     if #available(iOS 10.0, *) {
-      self.getNotificationAuthorizationStatus =
-        Signal.merge(
-          applicationLaunchedWithUser,
-          self.notificationAuthorizationCompletedProperty.signal)
+      self.getNotificationAuthorizationStatus = applicationLaunchedWithUser
 
-      self.registerForRemoteNotifications = self.notificationAuthorizationStatusProperty.signal
-        .skipNil()
-        .filter { $0.isAuthorized }
+      self.registerForRemoteNotifications = self.notificationAuthorizationCompletedProperty.signal
+        .filter(isTrue)
         .ignoreValues()
```

Comment:
Yeah, prompt is only shown once and then again if the app is deleted and reinstalled. From looking at how the code was written previously it appears that we try to cover all of the scenarios:

- App launched with options
- User session started
- Will enter foreground 

In the case of entering foreground a scenario could be: user denies permission, backgrounds the app, navigates to settings and grants permission and then when the app moves back to foreground we want to try to register again. I think in this particular scenario the permission would be `.granted` and not `.notDetermined`?